src/Nethermind/Nethermind.Api/IStateConfig.cs

```diff
@@ -0,0 +1,12 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using Nethermind.Config;
+
+namespace Nethermind.Api;
+
+public interface IStateConfig : IConfig
+{
+    [ConfigItem(Description = "How much state must be kept available from the head. Determine how old of a block can eth_call function. Also determine the reorg depth", DefaultValue = "64")]
+    int KeepLastNState { get; set; }
```

Comment:
put it to pruning config and call `PruningBoundry`?

Reply:
I'm trying to make it state db layout agnostic.

Reply:
Ok, but it is still related to state pruning

Reply:
Hmm.. hard to say. State may keep more and therefore have no pruning at all. FlatDB does not really prune anything, it just replace current keys. @damian-orzechowski what name do you want?

Reply:
Discarding old state can be considered (or just is) as pruning.

Reply:
I'll move this to pruning then.